### ISSUE TITLE
PasteRegistration not working fully correct for Extreme Mode

### ISSUE THREAD

**Issue by frudolph77** [alexander-zibert/turing-machine-board-game-solver#26]
Hey Alexander,

I've tried the PasteRegistraion with ~Nightmare~ Extreme Mode and stumbled upon errors, there where cards are missing.

Let's see this Hash `#E47 M9C` cards should be `A 25 3, B 4 12, C 5 19, D 17 20`. Shown  in any combination for a Verifier. 
```
#E47 M9C Share Get Criteria and Verification cards in the box.  A 325 346 B 412 614 C 519 233 D 2017 741 <- OK
                                            but could also be a ^ 32,5    ^ 41,2 
#E47 M9C Share Get Criteria and Verification cards in the box.  A 253 346 B 124 614 C 195 233 D 1720 741 <- NOK
                                                                2 53      1 24      1 95      17 20
#E47 M9C Share Get Criteria and Verification cards in the box.  A 253 346 B 124 614 C 519 233 D 2017 741 <- NOK
                                                                2 53      1 24      5 19      20 17
#E47 M9C Share Get Criteria and Verification cards in the box.  A 325 346 B 412 614 C 519 233 D 2017 741 <- OK
```
or even if you get all cards they are potentially wrong
```
#E49 R6K Share Get Criteria and Verification cards in the box.  A 1815 220 B 128 455 C 421 670 D 1310 632 <- OK
                                                                             ^ 1 28 instead of 12 8
```
Or try out these hashes `#E49 R6K`, `#E5V IRG` , `#E44 C5Q` 

There is no problem with 2 and 4 digit numbers, only for the three digit once. 
I think it could be possible to make it a little bit more fail save when checking the split numbers `XYZ` in combination of `XY` and `YZ`.
```
- YZ >= 49    => XY  and Z
- XY >= 49    => X  and YZ 

-  X >=  5    => X  and YZ
-  Y ==  0    => XY and Z
-  Z ==  0    => X  and YZ

every other numbers between X12 and X48 ,not matching the above criteria, are just guessing
```

I think it's impossible to make it working correct for nightmare mode, without an additional step and asking the player for the

**Comment by alexander-zibert** [comment]
Do you mean extreme mode instead of nightmare?
But you are right, the current implementation of the extreme mode parser depends on the criteria cards being sorted for one verifier.

However, it should be possible to differentiate both versions by the verification card. I'll look into this.

**Comment by alexander-zibert** [comment]
@frudolph77 The extreme mode parsing should work now for almost all cases. I included a map of all the cards with their corresponding verification mappings. This could be also used for verifying other input methods by checking that the verifier cards correspond to the crypt cards.

**Comment by frudolph77** [comment]
> Do you mean extreme mode instead of nightmare?

Yep, your right. :-)

**Comment by alexander-zibert** [comment]
No worries :)

I need to configure my notifications better. I only get notifications for new issues and pull requests and not for comments in existing issues / PRs.

**Comment by frudolph77** [comment]
can be closed on behalf of 1f052c7864a0f117dffae4dad519e3b69e908803

I tried the mentioned hash, seems to work fine.
